File: E-Commerce-Site/E-Commerce-Site.Cart/CartService.cs
using System.Xml;

namespace E_Commerce_Site.Cart;
using System.Data;
using System.Data.SqlClient;
using Models;
using Products.Models;
public class CartService : ICartService
{
    public async Task AddItemToCart(CartItem cart)
    {
        await using SqlConnection connection = new SqlConnection(Helper.CnnVal("EComm"));
        Guid uniqueId = Guid.NewGuid();

        await connection.OpenAsync();
        await using SqlCommand command = new SqlCommand("[addItemToCart]", connection);
        command.CommandType = CommandType.StoredProcedure;

        command.Parameters.AddWithValue("@cartId", cart.CartId);
        command.Parameters.AddWithValue("@userId", cart.UserId);
        command.Parameters.AddWithValue("@productId", cart.ProductId);
        command.Parameters.AddWithValue("@quantity", cart.Quantity);

        await command.ExecuteNonQueryAsync();
    }
    
    public async Task<List<CartProductCombo>> GetCartItems(string userId)
    {
        var results = new List<CartProductCombo>();
        await using (SqlConnection connection = new SqlConnection(Helper.CnnVal("EComm")))
        {
            SqlCommand command = new SqlCommand("dbo.get_cart_items", connection);
            command.CommandType = CommandType.StoredProcedure;
            command.Parameters.AddWithValue("@userId", userId);
            command.Connection.Open();
            SqlDataReader reader = await command.ExecuteReaderAsync();
            while (await reader.ReadAsync())
            {
                Console.WriteLine(reader[0]);
                Console.WriteLine(reader[1]);
                Console.WriteLine(reader[2]);
                Console.WriteLine(reader[3]);
                Console.WriteLine(reader[4]);
                Console.WriteLine(reader[5]);
                var path = reader[3] == DBNull.Value ? "" : (string)reader[3];
                results.Add( new CartProductCombo
                {
                    CartId = reader[0].ToString(),
                    Quantity = (int)reader[1],
                    Title = (string)reader[2],
                    ImagePath = path,
                    Retailer = (string)reader[4],
                    Cost = (decimal)reader[5],
                });
            }
            await reader.CloseAsync();
        }
        return results;
    }
    
    public async Task UpdateQuantity(CartIdClass cart)
    {
        await using SqlConnection connection = new SqlConnection(Helper.CnnVal("EComm"));
        await connection.OpenAsync();
        await using SqlCommand command = new SqlCommand("[alterQuantity]", connection);
        command.CommandType = CommandType.StoredProcedure;

        command.Parameters.AddWithValue("@cartId", cart.CartId);
        command.Parameters.AddWithValue("@quantity_change", cart.QuantityChange);

        await command.ExecuteNonQueryAsync();
    }
}

File: Python_BackEnd/Cart/CartService.py
from database import get_db_connection
from Cart.Models.Cart import CartItem, CartProductCombo, QuantityClass


async def add_item_to_cart(cart: CartItem) -> None:
    """
        Adds an item to the users cart
        Returns: None
   """
    command = "EXEC dbo.addItemToCart @cartId=?, @userId=?, @productId=?, @quantity=?"
    params = (
        cart.cartId,
        cart.userId,
        cart.productId,
        cart.quantity
    )
    db = get_db_connection()
    with db.cursor() as cursor:
        cursor.execute(command, params)


async def get_cart_items(user_id: str) -> list[CartProductCombo]:
    """
        Gets all items in the cart for a given userID
        Returns: list[CartProductCombo] A list of all cart items in a given users cart
   """
    command = "EXEC dbo.get_cart_items @userId=?"
    params = user_id
    db = get_db_connection()
    with db.cursor() as cursor:
        cursor.execute(command, params)
        results = [CartProductCombo(*row) for row in cursor]
    return results


async def update_quantity(cart: QuantityClass) -> None:
    """
        Update the quantity of an item in the cart
        Returns: None
   """
    command = "EXEC dbo.alterQuantity @cartId=?, @quantity_change=?"
    params = (cart.cartId, cart.quantityChange)
    db = get_db_connection()
    with db.cursor() as cursor:
        cursor.execute(command, params)


async def remove_item(cart_id: str) -> None:
    """
        Remove an item in the cart
        Returns: None
   """
    command = "EXEC dbo.remove_item @cartId=?"
    params = cart_id
    db = get_db_connection()
    with db.cursor() as cursor:
        cursor.execute(command, params)
